lib/beacon/content/js_hook.ex

```diff
@@ -0,0 +1,50 @@
+defmodule Beacon.Content.JSHook do
+  @moduledoc """
+  Stores a JS Hook which can be referenced from your Beacon pages, layouts, and components.
+
+  > #### Do not create or edit JS Hooks manually {: .warning}
+  >
+  > Use the public functions in `Beacon.Content` instead.
+  > The functions in that module guarantee that all dependencies
+  > are created correctly and all processes are updated.
+  > Manipulating data manually will most likely result
+  > in inconsistent behavior and crashes.
+
+  """
+
+  use Beacon.Schema
+
+  @type t :: %__MODULE__{}
+
+  schema "beacon_js_hooks" do
+    field :name, :string
+    field :site, Beacon.Types.Site
+
+    field :mounted, :string
+    field :beforeUpdate, :string
```

Comment:
Names in Elixir and DB in general are in the snake case format, shouldn't be better to follow that pattern here?
```suggestion
    field :before_update, :string
```

Reply:
my intention here was that formatting the callback name would be simple `to_string(name)` since the JS Hook syntax expects camelCase.  I suppose we could switch the field name to snake_case, and then add an additional formatting step like:
```elixir
[head | tail] = String.split(name, "_")
tail = Enum.map(tail, &String.capitalize/1)
Enum.join([head | tail])
```